addons/settings/fnc_gui_createCategory.sqf

```diff
@@ -111,27 +113,11 @@ private _lastSubCategory = "$START";
         };
 
         // ----- create setting group
-        private _ctrlSettingGroup = switch (toUpper _settingType) do {
-            case "CHECKBOX": {
-                _display ctrlCreate [QGVAR(Row_Checkbox), IDC_SETTING_CONTROLS_GROUP, _ctrlOptionsGroup]
-            };
-            case "EDITBOX": {
-                _display ctrlCreate [QGVAR(Row_Editbox), IDC_SETTING_CONTROLS_GROUP, _ctrlOptionsGroup]
-            };
-            case "LIST": {
-                _display ctrlCreate [QGVAR(Row_List), IDC_SETTING_CONTROLS_GROUP, _ctrlOptionsGroup]
-            };
-            case "SLIDER": {
-                _display ctrlCreate [QGVAR(Row_Slider), IDC_SETTING_CONTROLS_GROUP, _ctrlOptionsGroup]
-            };
-            case "COLOR": {
-                _display ctrlCreate [[QGVAR(Row_Color), QGVAR(Row_ColorAlpha)] select (count _defaultValue > 3), IDC_SETTING_CONTROLS_GROUP, _ctrlOptionsGroup]
-            };
-            case "TIME": {
-                _display ctrlCreate [QGVAR(Row_Time), IDC_SETTING_CONTROLS_GROUP, _ctrlOptionsGroup]
-            };
-            default {controlNull};
+        private _settingTypeTemp = _settingType;
+        if (_settingTypeTemp == "COLOR" && {count _defaultValue > 3}) then {
```

Comment:
```suggestion
        if (_settingTypeTemp isEqualTo "COLOR" && {count _defaultValue > 3}) then {
```

Reply:
Again this, why?

Reply:
Everything stacks. If neither the intended function or the performance doesn't change for the worse, then nothing is lost; small gains, no matter how small, are always welcome.

Reply:
Need to make sure `_settingTypeTemp` is always uppercase then. But my concern with this wasn't performance. Just reducing lines/making them easier on the eyes.

Reply:
There is one loss, readability. I don't see any gains, even small onesy though.

Reply:
Hardly any readability loss.

If the casing is questionable, then there should be a comment, and wherever the problem source for the questionable casing is needs to be fixed.

Reply:
It also loses robustness, because IET silently returns false for mismatched types, when it really should blow up instead.

Reply:
I suppose that's a fair point.